packages/pangea-multipass/pangea_multipass/sources/slack/slack.py

```diff
@@ -21,19 +25,110 @@ def list_channels(token: str):
             return channels
         except SlackApiError as e:
-            print(f"Error listing channels: {e.response['error']}")
             return []
 
+    @staticmethod
+    def get_channel_members(token: str, channel_id: str):
+        """
+        Retrieve the list of members in a Slack channel.
+
+        Args:
+            token (str): Slack token.
+            channel_id (str): Channel id to request members.
+
+        Returns:
+            List of user IDs in the channel.
+        """
+
+        client = WebClient(token=token)
+        try:
+            response = client.conversations_members(channel=channel_id)
+            return response["members"]
+        except SlackApiError as e:
+            return None
+
+    @staticmethod
+    def get_all_channels(token: str):
+        """
+        Retrieve all channels in the workspace.
+
+        Args: 
+            token (str): Slack token
+
+        Returns:
+            List of channel IDs.
+        """
+
+        client = WebClient(token=token)
+        channels = []
+        try:
+            response = client.conversations_list(types="public_channel,private_channel", limit=1000)
+            channels = response.get("channels", [])
+            return [channel["id"] for channel in channels]
+        except SlackApiError as e:
+            return []
+
+    @staticmethod
+    def get_user_id(token: str, user_email: str):
+        """
+        Retrieve the Slack user ID for a given email address.
+
+        Args:
+            token (str): Slack token.
+            user_email (str): User email to request user id.
+
+        Returns:
+            User ID or None if the user does not exist.
+        """
+
+        client = WebClient(token=token)
+        try:
+            response = client.users_lookupByEmail(email=user_email)
+            return response["user"]["id"]
+        except SlackApiError as e:
+            return None
+
+    @staticmethod
+    def get_channels_for_user(token: str, user_id: str, channel_ids: List[str]):
+        """
+        Check which channels a user has access to.
+
+        Args:
+            token (str): Slack token.
+            user_id (str): Slack user id.
+            channels_ids (List[str]): Channels id to check access for user_id.
+
+        Returns:
+            List of channel IDs the user has access to.
+        """
+        client = WebClient(token=token)
+        accessible_channels = []
+        for channel_id in channel_ids:
+            try:
+                response = client.conversations_members(channel=channel_id)
+                members = response.get("members", [])
+                if user_id in members:
+                    accessible_channels.append(channel_id)
+            except SlackApiError as e:
+                if e.response["error"] == "not_in_channel":
+                    continue  # User is not in this channel
+                else:
+                    print(f"Error checking channel {channel_id}: {e.response['error']}")
```

Comment:
Do we want to keep this `print()` here?

Reply:
Hmm, I'll remove it and add a ticket to add a `logger` to the whole package.